src/common.rs

```diff
@@ -64,6 +63,24 @@ pub fn set_fd_close_exec(fd: RawFd) -> Result<RawFd> {
     Ok(fd)
 }
 
+// SOCK_CLOEXEC flag is Linux specific
+#[cfg(target_os = "linux")]
+pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::SOCK_CLOEXEC;
+#[cfg(not(target_os = "linux"))]
+pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::empty();
+
+#[cfg(target_os = "linux")]
+fn make_addr(host: &str) -> Result<UnixAddr> {
+    UnixAddr::new_abstract(host.as_bytes()).map_err(err_to_others_err!(e, ""))
+}
+
+#[cfg(not(target_os = "linux"))]
+fn make_addr(host: &str) -> Result<UnixAddr> {
+    UnixAddr::new(host).map_err(err_to_others_err!(e, ""))
```

Comment:
I think we should detect and delete the already exist address to avoid the error EADDRINUSE

Reply:
Let's do a follow up PR for this?
Not exactly in scope of this one.

Reply:
fine.